ChangeLog

```diff
@@ -35,17 +35,31 @@ MODS:
 * Added bonus updaters for hero specialties
 * Added allOf, anyOf and noneOf qualifiers for bonus limiters
 * Added bonus limiters: alignment, faction and terrain
+* Supported new terrains, new battlefields, custom water and rock terrains
 

```

Comment:
Adventure map:
* added buttons and keyboard shortcuts to quickly exchange army and artifacts between heroes